### PR TITLE
Studio service and controller

### PR DESCRIPTION
**Pull request by Rangel-Plachkov**

### CODE REVIEW

File: PopcornPals-Server/src/main/java/bg/fmi/popcornpals/controller/StudioController.java
Comment (denib13) [inline PopcornPals-Server/src/main/java/bg/fmi/popcornpals/controller/StudioController.java:1]: correct me if i am wrong, but in my opinion and also based on what i've seen from other demos, shouldn't you write all the logic behind the CRUD operations in the service and just use it here by simply calling the methods in the service?

Comment (denib13) [reply PopcornPals-Server/src/main/java/bg/fmi/popcornpals/controller/StudioController.java:1]: i mean basically what you did in the service should be done in here and vice versa

Comment (denib13) [reply PopcornPals-Server/src/main/java/bg/fmi/popcornpals/controller/StudioController.java:1]: and one last thing, i think it is quite a standard @RequestMapping("/studios") to be @RequestMapping("/api/studios") (not sure if there's a slash before api or not)

**Review by denib13 — COMMENTED** [review]
check my comments on the controller file and let's discuss them first before merging :)

File: PopcornPals-Server/src/main/java/bg/fmi/popcornpals/controller/StudioController.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    private StudioService studioService;
+    @Autowired
+    private StudioMapper studioMapper;
+
+    @PostMapping("/create")
+    public ResponseEntity<StudioDTO> createStudio(@Valid @RequestBody StudioDTO studio) {
+        Studio createdStudio = studioService.createStudio(studioMapper.toEntity(studio));
+        return new ResponseEntity<>(studioMapper.toDTO(createdStudio), HttpStatus.CREATED);
+    }
+    @GetMapping("/readById/{id}")
```
Comment (denib13) [inline PopcornPals-Server/src/main/java/bg/fmi/popcornpals/controller/StudioController.java]: not a RESTful endpoint, i suggest leaving it just to "/{id}"

File: PopcornPals-Server/src/main/java/bg/fmi/popcornpals/controller/StudioController.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    public ResponseEntity<StudioDTO> getStudioById(@PathVariable Long id) {
+        Studio studio = studioService.getStudioById(id);
+        if (studio == null) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+        return new ResponseEntity<>(studioMapper.toDTO(studio), HttpStatus.OK);
+    }
+    @GetMapping("/readByName")
```
Comment (denib13) [inline PopcornPals-Server/src/main/java/bg/fmi/popcornpals/controller/StudioController.java]: not a RESTful endpoint, i suggest leaving it just to "/{name}"

File: PopcornPals-Server/src/main/java/bg/fmi/popcornpals/controller/StudioController.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    public ResponseEntity<List<StudioDTO>> getStudioByName(@RequestParam (name = "name") String name) {
```
Comment (denib13) [inline PopcornPals-Server/src/main/java/bg/fmi/popcornpals/controller/StudioController.java]: name must be path variable

File: PopcornPals-Server/src/main/java/bg/fmi/popcornpals/dto/StudioDTO.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import lombok.NoArgsConstructor;
+
+import java.time.LocalDate;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class StudioDTO {
+
+    @NotNull(message = "Studio: Name cannot be null")
```
Comment (denib13) [inline PopcornPals-Server/src/main/java/bg/fmi/popcornpals/dto/StudioDTO.java:22]: No need for validation in the DTOs

Comment (Rangel-Plachkov) [reply PopcornPals-Server/src/main/java/bg/fmi/popcornpals/dto/StudioDTO.java:22]: As far as i am aware its more important for the DTO to have validatian rather thatn the model

**Pull request merged**